apps/provider-console/src/components/become-provider/ProviderAttributes.tsx

```diff
@@ -0,0 +1,170 @@
+"use client";
+
+import React from "react";
+import { Controller, SubmitHandler, useFieldArray, useForm } from "react-hook-form";
+import {
+  Button,
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormMessage,
+  Input,
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  Separator
+} from "@akashnetwork/ui/components";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useAtom } from "jotai";
+import { PlusIcon, TrashIcon } from "lucide-react";
+import { z } from "zod";
+
+import providerProcessStore from "@src/store/providerProcessStore";
+import { providerAttributesFormValuesSchema } from "../../types/providerAttributes";
+import { ResetProviderForm } from "./ResetProviderProcess";
+
+const attributeKeys = Object.keys(providerAttributesFormValuesSchema.shape);
+
+interface ProviderAttributesProps {
+  stepChange: () => void;
+}
+
+const providerFormSchema = z.object({
+  attributes: z.array(
+    z.object({
+      key: z.string().min(1, "Key is required"),
+      value: z.string().min(1, "Value is required"),
+      customKey: z.string().optional()
+    })
+  )
+});
+
+type ProviderFormValues = z.infer<typeof providerFormSchema>;
+
+export const ProviderAttributes: React.FC<ProviderAttributesProps> = ({ stepChange }) => {
+  const [providerPricing, setProviderPricing] = useAtom(providerProcessStore.providerProcessAtom);
+  const form = useForm<ProviderFormValues>({
+    resolver: zodResolver(providerFormSchema),
+    defaultValues: {
+      attributes: [{ key: "", value: "", customKey: "" }]
+    }
+  });
+
+  const { control } = form;
+
+  const { fields, append, remove } = useFieldArray({
+    control,
+    name: "attributes"
+  });
+
+  const onSubmit: SubmitHandler<ProviderFormValues> = async data => {
```

Comment:
function names should reflect what they are doing, not when they are called. The main reason: if I read the code 

```tsx
<form onSubmit={form.handleSubmit(onSubmit)} className="space-y-6">
```

I have no idea what happens on submit, I need to go and read the code. Instead if it was

```tsx
<form onSubmit={form.handleSubmit(updatePricingAndProceed)} className="space-y-6">
```

it would help to reduce cognitive load and perhaps leave potential dev reading the code satisfied with that knowledge.

Reply:
changed to onComplete as submission and handling data happened before we go to next step.